tests/OmniSharp.Roslyn.CSharp.Tests/FormatAfterKeystrokeTests.cs

```diff
@@ -360,6 +360,106 @@ public void M()
     {
         ///
 $$
+    }
+}");
+        }
+
+        [Theory]
+        [InlineData("file.cs")]
+        [InlineData("file.csx")]
+        public async Task LeadingSpacesNotRemovedOnNewline_ExtraWhitespaceOnPreviousLine(string fileName)
+        {
+            await VerifyChange(fileName, "\n",
+@"class C
+{
+    public void M()
+    {
+        // There is a space after the ; here
+        var i = 1; 
+        $$
+    }
+}",
+@"class C
+{
+    public void M()
+    {
+        // There is a space after the ; here
+        var i = 1;
+        
+    }
+}");
+        }
+
+        [Theory]
+        [InlineData("file.cs")]
+        [InlineData("file.csx")]
+        public async Task LeadingSpacesNotRemovedOnNewline_FormattingChangesOnPreviousLine(string fileName)
+        {
+            await VerifyChange(fileName, "\n",
+@"class C
+{
+    public void M()
+    {
+        var i =1;
+        $$
+    }
+}",
+@"class C
+{
+    public void M()
+    {
+        var i = 1;
+        
+    }
+}");
+        }
+
+        [Theory]
+        [InlineData("file.cs")]
+        [InlineData("file.csx")]
+        public async Task LeadingSpacesNotRemovedOnNewline_NewlinesWithSpaceAfter(string fileName)
+        {
+            await VerifyChange(fileName, "\n",
+@"class C
+{
+    public void M()
+    {
+        var i =1;
+        $$
+        
+    }
+}",
+@"class C
+{
+    public void M()
+    {
+        var i = 1;
+        
+        
+    }
+}");
+        }
+
+        [Theory(Skip = "https://github.com/dotnet/roslyn/issues/50130")]
```

Comment:
I'm not sure there's a good way to work around this one until the upstream bug is fixed.